internal/collector/processor/nodebatcher/node_batcher.go

```diff
@@ -237,13 +237,18 @@ func (nb *nodeBatcher) add(spans []*tracepb.Span) {
 
 // cutBatch cuts the provided batch, and sets a new batch on this nodeBatcher
 func (nb *nodeBatcher) cutBatch(b *batch) {
+	initialCap := b.getCurrentCap()
+	currSize := b.getCurrentItemCount()
+	for currSize < initialCap>>1 && initialCap>>1 > initialBatchCapacity {
```

Comment:
nit: adding a for loop local (`newInitialCap` or `reducedCap` perhaps?) makes easier to see what the loop is doing, and avoids repeating the `>>`  operation